### ISSUE TITLE
Add missing `shape_fcn` and `shifted_shape_fcn` methods for MasterElements

### ISSUE THREAD

**Issue by sayerhs** [Exawind/nalu-wind#184]
Before transitioning Kernels to run on GPUs, at least the following MasterElements must have implementations of `shape_fcn` and `shifted_shape_fcn` that take `SharedMemView<...>` arguments:

- [x] Hex8SCV
- [x] Quad42DSCV
- [x] Quad42DSCS
- [x] Tet4SCV
- [x] Tet4SCS
- [x] Wedge6SCV
- [x] Wedge6SCS

**Comment by sayerhs** [comment]
The following cases need to be fixed 

```
The following tests FAILED:
	 15 - dgNonConformalElemCylinder (Failed)
	 18 - dgNonConformalThreeBlade (Failed)
	 19 - ductElemWedge (Failed)
	 27 - elemHybridFluids (Failed)
	 28 - elemHybridFluidsShift (Failed)
	 35 - heatedWaterChannelElem (Failed)
	 48 - nonConformalWithPeriodicConsolidated (Failed)
	 71 - dgncThreeBladeHypre (Failed)
```

**Comment by sayerhs** [comment]
Addressed in commits: 49423d34, 9cc7e7f3ab6, 2340eb91a43, and 9932225a3d7
